### PR TITLE
storage: fix grants for tarantool 3 config cluster

### LINKED ISSUE #412: `box.info.ro` can be true in `init_storage()` on all instances in replicaset
As the result, access grants may be skipped.

This should be documented or the code should account for it.

### PR DESCRIPTION
**Pull request by DifferentialOrange**
Fix vshard user grants setup for Tarantool 3.x through async bootstrap. See commits for more info.

I didn't forget about

- [x] Tests
- [x] Changelog
- [x] Documentation

Closes #412
Part of #415

### CODE REVIEW

File: crud/storage.lua [hunk 1]
```diff
@@ -0,0 +1,145 @@
+local checks = require('checks')
+local dev_checks = require('crud.common.dev_checks')
+local stash = require('crud.common.stash')
```
Comment (better0fdead) [inline crud/storage.lua:3]: NIT: divide 'checks' require, from 'crud.' requires.

**Review by better0fdead — APPROVED** [review]
Hi, thx for patch

File: crud/storage.lua (hunk 1 above)
Comment (DifferentialOrange) [reply crud/storage.lua:4]: Updated

File: test/helper.lua [hunk 2]
```diff
@@ -1254,4 +1254,16 @@ function helpers.wait_schema_init()
...
+    return crud_utils.tarantool_supports_box_watch()
+end
+
+function helpers.skip_box_watch_unsupported()
+    t.skip_if(not helpers.is_box_watch_supported(), 'box.watch is not supported')
+end
+
+function helpers.skip_box_watch_supported()
+    t.skip_if(helpers.is_box_watch_supported(), 'box.watch is supported')
+end
```
Comment (oleg-jukovec) [inline test/helper.lua:1267]: It is less confusing, but up to you.
```suggestion
function helpers.skip_if_box_watch_unsupported()
    t.skip_if(not helpers.is_box_watch_supported(), 'box.watch is not supported')
end

function helpers.skip_if_box_watch_supported()
    t.skip_if(helpers.is_box_watch_supported(), 'box.watch is supported')
end
```

File: crud/storage_info.lua [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    local vshard_router, err = utils.get_vshard_router_instance(opts.vshard_router)
+    if err ~= nil then
+        return nil, StorageInfoError:new(err)
+    end
+
+    local replicasets, err = vshard_router:routeall()
+    if replicasets == nil then
+        return nil, StorageInfoError:new("Failed to get router replicasets: %s", err.err)
+    end
```
Comment (oleg-jukovec) [inline crud/storage_info.lua:54]: O am I missing something?

https://www.tarantool.io/ru/doc/latest/reference/reference_rock/vshard/vshard_router/#lua-function.vshard.router.routeall
```suggestion
    local replicasets, err = vshard_router:routeall()
    if err ~= nil then
        return nil, StorageInfoError:new("Failed to get router replicasets: %s", err.err)
    end
```

File: crud/storage_info.lua [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    local futures_by_replicas = {}
+    local replica_state_by_id = {}
+    local async_opts = {is_async = true}
+    local timeout = opts.timeout or const.DEFAULT_VSHARD_CALL_TIMEOUT
+
+    for _, replicaset in pairs(replicasets) do
+        for replica_id, replica in pairs(replicaset.replicas) do
+            local master = utils.get_replicaset_master(replicaset, {cached = false})
+
```
Comment (oleg-jukovec) [inline crud/storage_info.lua:64]: What is there an idea to get master on each iteration?

File: test/helper.lua (hunk 2 above)
Comment (DifferentialOrange) [reply test/helper.lua]: Updated

File: crud/storage_info.lua (hunk 3 above)
Comment (DifferentialOrange) [reply crud/storage_info.lua]: Yeah, it seems that we had missed it on the original PR review. Updated. This commit only moves existing code, so I had put improvements to a separate commit "storage: refactor replicasets process on info".

File: crud/storage_info.lua (hunk 4 above)
Comment (DifferentialOrange) [reply crud/storage_info.lua]: It seems that's one more thing we had missed on the original PR review. Updated. This commit only moves existing code, so I had put improvements to a separate commit "storage: refactor replicasets process on info".

**Review by oleg-jukovec — APPROVED** [review]

**Pull request merged**